src/reporting/Market.sol

```diff
@@ -455,13 +456,17 @@ contract Market is DelegationTarget, Typed, Initializable, Ownable, IMarket {
             return ReportingState.ALL_REPORTING;
         }
 
-        // Either no automated report was made or the automated report was disputed so we are in some phase of limited reporting
-        if (block.timestamp > reportingWindow.getEndTime()) {
+        if (_reportingWindowOver) {
             return ReportingState.AWAITING_FINALIZATION;
         }
 
+        // Either no automated report was made or the automated report was disputed so we are in some phase of limited reporting
         if (reportingWindow.isDisputeActive()) {
-            return ReportingState.LIMITED_DISPUTE;
+            if (tentativeWinningPayoutDistributionHash == bytes32(0)) {
+                return ReportingState.AWAITING_FINALIZATION;
```

Comment:
I believe we discussed this previously in this PR, but I don't remember the outcome.  Shouldn't this be `AWAITING_MIGRATION` or something as this market needs to move to a future reporting window?

Reply:
Since we no longer need to check if any reports were made I was able to remove the explicit status for finalization migration in favor of just checking the tentative payout hash. Recall before we had a new status for this `AWAITING_FINALIZATION_MIGRATION`

Reply:
But this market isn't actually awaiting finalization right?  It is awaiting being moved to a new reporting window?

Reply:
Technically it will be migrated but the awaiting migration status is actually pretty specific to forking. We could rename that status to something more specific `AWAITING_FORK_MIGRATION` and then add back `AWAITING_FINALIZATION_MIGRATION`

Reply:
It isn't awaiting finalization at all though.  It is awaiting migration to a new reporting window.  My contention is primarily around the `FINALIZATION` bit, since that implies the market is ready to finalize (which it most definitely isn't).

`AWAITING_NO_REPORT_MIGRATION` perhaps?  This would align with a change to `AWAITING_FORK_MIGRATION` where the name describes the type of migration we are waiting on.

Alternatively, we could include the migration destination instead:
`AWAITING_UNIVERSE_MIGRATION` vs `AWAITING_REPORTING_WINDOW_MIGRATION`.

Reply:
I like the `AWAITING_NO_REPORT_MIGRATION` and `AWAITING_FORK_MIGRATION`.

Do we want to make a new function to trigger the `NO REPORT` migration or continue to do it when `tryFinalize` is called?

Reply:
Can we make the move happen automatically when someone tries to report on the market?

Reply:
Time-wise it will always be after reporting for the market is finished so that seems like an odd way to trigger it to me unless I'm missing something.

Reply:
1. Market is in window 5, receives no reports.
2. Time passes and window 6 begins.
3. UI presents market to user as part of window 6.
4. User attempts to report on market (whose state is still AWAITING_NO_REPORT_MIGRATION`).
5. As part of the report transaction, the market is moved to window 6 (which will change its state to `LIMITED_REPORTING`).

Reply:
I get the goal though, to have there be no interruption and seamlessly be able to be reported on. We could do this as long as we think it makes sense to consumers (API mostly)

Reply:
I think this is a solvable problem at the UI level, and from the contracts this is the best option as it doesn't require a user signing a one-off transaction just to trigger a move so they can then report on it.  Things do get weird if a fork starts when a market is in this state, so we'll need to make sure to handle that.